File: recursion/reverse_string.py
# reverse string 

def reverse_string(s):
    rev_string = ""

    for str in s:
        rev_string = str + rev_string
    
    return rev_string


# print(reverse_string("hello"))

def reverse_string_recursive(s):
    if len(s) <= 1:
        return s
    
    return reverse_string_recursive(s[1:]) + s[0]


print(reverse_string_recursive("tiger"))

File: recursion/reverse_string.js
// classic approach

function reverse_string(s) {
  rev_str = "";

  for (str of s) {
    rev_str = str + rev_str;
  }

  return rev_str;
}

// console.log(reverse_string("hello"));

function reverse_string_recursive(s) {
  if (s.length <= 1) return s;

  return reverse_string_recursive(s.slice(1)) + s[0];
}

console.log(reverse_string_recursive("tiger"));
